bin/dasht

```diff
@@ -67,11 +72,29 @@
 # Written in 2016 by Suraj N. Kurapati <https://github.com/sunaku/dasht>
 # Distributed under the terms of the ISC license (refer to README file).
 
+: ${DASHT_USE_DASH_APP:=false}
+if $DASHT_USE_DASH_APP; then
+  : ${DASHT_DOCSETS_DIR:="$HOME/Library/Application Support/Dash"}
+else
+  : ${DASHT_DOCSETS_DIR:=${XDG_DATA_HOME:-$HOME/.local/share}/dasht/docsets}
+fi
+
 if test $# -gt 0; then
   pattern=$1
   shift
 fi
 
+# Ensure any docsets we are accessing from the Dash.app library have had their
+# documents extracted. Dash.app leaves the documents in the tarball by default.
+if $DASHT_USE_DASH_APP; then
+  dasht-docsets "$@" | while read docset; do
+    root=("$DASHT_DOCSETS_DIR/"*/*"/${docset}.docset")
+    if ! test -d "${root}/Contents/Resources/Documents"; then
+      dasht-docsets-extract "$root/Contents/Resources/tarix.tgz"
```

Comment:
I think this extraction would better fit in a different script, say `dasht-docsets-inherit`, which would focus on preparing docsets from other apps for use with dasht.  This way, the `dasht` script won't exceed its original purpose.

Reply:
OK, I made the requested changes and tested it out a bit, I think all is working well. BTW, you should update gitignore and instructions re manpages:

- `/man` is in gitignore despite being included in the repo
- I am not familiar with `binman-rake`, but I think it's behavior changed since you posted the instructions. I ran it and it generated a bunch of html, markdown, and a CSS flie. I deleted all of the excess files it generated and just included the changes to the manpage-format one and the `dasht-docsets-inherit`.

Reply:
Thanks for your feedback.  Actually, both of those are exactly as they're meant to be:
* For gitignore, `man/` is ignored (line 2) but some of its contents aren't (line 1). :nerd_face:
* For `binman-rake`, the excess files it generates are ignored by gitignore line 1.

```gitignore
!/man/man?/*.?
/man
```